nodebb

```diff
@@ -2,4 +2,11 @@
 
 'use strict';
 
-require('./src/cli');
+// This permits to have 2 differents entrypoints
+// with the same interface nodebb
+
+if (process.argv[2] === 'cmd') {
```

Comment:
Did you consider using a commander subcommand instead of messing with argv directly?

If you did that you wouldn't have to do any duplicate bootstrapping.

Reply:
Of course, I considered it as the first solution.
The problem is not so easy, because all plugin commands must be registered after asynchronously after active plugins are loaded...
So I took the decision to create a second entrypoint that separate the standard ./nodebb logic, without doing mess inside it.
Suddenly we have a double bootstrap...
But I consider to have 2 differents bootstrap in the near future, one that includes the HTTP layer logic (routes, templates etc) and one that is lighter CLI-only.
Any better idea?

Reply:
I'm pretty sure you could add a new `cmd` Command to the existing commander Program, then create a brand new commander Program on demand within that command, passing it the given arguments.

Reply:
Yep. You'll want the plugin prefix removal thing before you create argv, and I'd probably just pull all of that into a separate function in a module under `src/cli`

Reply:
Here I simplified the module https://github.com/NodeBB/NodeBB/pull/6901/commits/63ffaa9f85d6f8d16659565d0d39836d99a8b4c6